src/main/java/com/pgs/booking/service/WeatherService.java

```diff
@@ -0,0 +1,57 @@
+package com.pgs.booking.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pgs.booking.exception.ResourceNotFoundException;
+import com.pgs.booking.model.dto.WeatherDto;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Getter
+@Setter
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WeatherService {
+
+    private final String keyValue = "ee6bbf09dcc0ab10e314eda8dce3dc74";
+    private final String mode = "json";
+    private final String units = "imperial";
+    private final ObjectMapper objectMapper;
+
+    public WeatherDto getWeatherByCity(String city) {
+        RestTemplate restTemplate = new RestTemplate();
```

Comment:
create restTeamplate in a new Config class (see annotation @Configuration and @Bean)
and finally inject it like ObjectMapper

Reply:
ok, i will change it